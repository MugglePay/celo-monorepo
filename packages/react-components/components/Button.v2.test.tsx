import Button, { BtnTypes } from '@celo/react-components/components/Button.v2'
import * as React from 'react'
import { Text } from 'react-native'
import { fireEvent, render } from 'react-native-testing-library'
import * as renderer from 'react-test-renderer'

describe('Button', () => {
  describe('when pressed', () => {
    it('fires the onPress prop', () => {
      const handler = jest.fn()
      const { getByTestId } = render(
        <Button onPress={handler} text="Button" type={BtnTypes.PRIMARY} testID={'TEST'} />
      )
      const button = getByTestId('TEST')

      fireEvent.press(button)
      expect(handler).toBeCalled()
    })

    it('multiple times fires once', () => {
      const handler = jest.fn()
      const { getByTestId } = render(
        <Button onPress={handler} text="Button" type={BtnTypes.PRIMARY} testID={'TEST'} />
      )
      const button = getByTestId('TEST')
      fireEvent.press(button)
      expect(handler).toBeCalledTimes(1)
    })
  })

  it('renders with minimum props', () => {
    const button = render(<Button onPress={jest.fn()} text="Button" type={BtnTypes.PRIMARY} />)
    expect(button.getByText('Button')).toBeTruthy()
  })
  describe('when children passed', () => {
    it('renders with children and text', () => {
      const { queryByTestId, queryByText } = render(
        <Button onPress={jest.fn()} text="Button" type={BtnTypes.PRIMARY}>
          <Text testID="Child">child text</Text>
        </Button>
      )
      expect(queryByTestId('Child')?.props.children).toEqual('child text')
      expect(queryByText('Button')).toBeTruthy()
    })
  })
  describe('when passed accessibilityLabel', () => {
    it('sets it', () => {
      const { getByA11yLabel } = render(
        <Button
          accessibilityLabel="link"
          onPress={jest.fn()}
          text="Button"
          type={BtnTypes.PRIMARY}
        />
      )
      expect(getByA11yLabel('link').children).toContain('Button')
    })
  })
  describe('when type is SECONDARY', () => {
    it('renders', () => {
      const tree = renderer.create(
        <Button onPress={jest.fn()} text="Button" type={BtnTypes.SECONDARY}>
          <Text>child text</Text>
        </Button>
      )
      expect(tree).toMatchSnapshot()
    })
  })
  describe('when type is TERTIARY', () => {
    it('renders', () => {
      const tree = renderer.create(
        <Button onPress={jest.fn()} text="Button" type={BtnTypes.TERTIARY}>
          <Text>child text</Text>
        </Button>
      )
      expect(tree).toMatchSnapshot()
    })
  })
})
