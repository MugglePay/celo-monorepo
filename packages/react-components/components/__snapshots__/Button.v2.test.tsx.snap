// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when type is SECONDARY renders 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    nativeBackgroundAndroid={
      Object {
        "attribute": "selectableItemBackground",
        "type": "ThemeAttrAndroid",
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 100,
          "justifyContent": "center",
          "margin": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 5,
        },
        Object {
          "height": 48,
          "minWidth": 98,
        },
        Object {
          "backgroundColor": "#F5F4F0",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <Text>
        child text
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 15,
              "lineHeight": 20,
            },
            Object {
              "color": "#2E3338",
            },
          ]
        }
      >
        Button
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button when type is TERTIARY renders 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    nativeBackgroundAndroid={
      Object {
        "attribute": "selectableItemBackground",
        "type": "ThemeAttrAndroid",
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 100,
          "justifyContent": "center",
          "margin": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 5,
        },
        Object {
          "height": 48,
          "minWidth": 98,
        },
        Object {
          "backgroundColor": "#EEB93C",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <Text>
        child text
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 15,
              "lineHeight": 20,
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        Button
      </Text>
    </View>
  </View>
</View>
`;
